src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -127,35 +126,44 @@ private void DeterminePropertyName(JsonSerializerOptions options)
                 // Set the compare name.
                 if (options.PropertyNameCaseInsensitive)
                 {
-                    CompareNameAsString = NameAsString.ToUpperInvariant();
-                    _compareName = Encoding.UTF8.GetBytes(CompareNameAsString);
+                    NameUsedToCompareAsString = NameAsString.ToUpperInvariant();
+                    _nameUsedToCompare = Encoding.UTF8.GetBytes(NameUsedToCompareAsString);
                 }
                 else
                 {
-                    CompareNameAsString = NameAsString;
-                    _compareName = _name;
+                    NameUsedToCompareAsString = NameAsString;
+                    _nameUsedToCompare = _name;
                 }
 
                 // Cache the escaped name.
+                _escapedName = _name; // temporary behavior until the reader can accept escaped string.
+#if false
                 int valueIdx = JsonWriterHelper.NeedsEscaping(_name);
                 if (valueIdx == -1)
                 {
                     _escapedName = _name;
                 }
                 else
                 {
+                    byte[] pooledName = null;
                     int length = JsonWriterHelper.GetMaxEscapedLength(_name.Length, valueIdx);
 
-                    byte[] tempArray = ArrayPool<byte>.Shared.Rent(length);
+                    Span<byte> escapedName = length <= JsonConstants.StackallocThreshold ?
+                        stackalloc byte[length] :
```

Comment:
length check should be uint to help guard against negative values.

Reply:
Agree but it looks like a pervasive problem - the API being called should probably be uint.

uint is not CLS compliant, so that explains the public APIs anyway.

Reply:
If you think this is necessary, should we be doing that everywhere else as well? We have this sort of pattern in a lot of places.

https://github.com/dotnet/corefx/blob/be77af0a58a0ea8399f93d86d3338f10095df104/src/Common/src/CoreLib/System/Globalization/IdnMapping.Unix.cs#L25-L27

https://github.com/dotnet/corefx/blob/9e2f3b630a51875fc0455a7cabb8803f6527f81e/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs#L640-L644